src/fsharp/pars.fsy

```diff
@@ -930,8 +938,14 @@ tyconSpfnRhs:
            SynTypeDefnSig(nameInfo, mEquals, SynTypeDefnSigRepr.Simple ($1, $1.Range), None, augmentation, mWhole)) }
 
   | tyconClassSpfn 
-     { let objectModelRange = lhs parseState 
-       let needsCheck, (kind, decls) = $1
+     {  let needsCheck, (kind, decls) = $1
+        let objectModelRange =
+            match decls with
+            | [] -> lhs parseState
```

Comment:
@nojaf Would it be hard to extract successfully ranges here too? Sorry, I should've added comment here too. 🙂

Reply:
Would that be replacing it with `rhs parseState 1`?

Reply:
From a glance, it seems it may require passing parsed ranges via `tyconClassSpfn` and other rules. Otherwise recovered ranges may still span further than they should. Not sure if you'd want to do that as a part of this PR. 🙂

It'd be great to do this systematically, the same as we've been fixing ranges of the things.

Reply:
Hmm, I see what you mean. I'm not going to push my luck in this current PR 😅.
I'll let it be while the PR is still green.

Reply:
@nojaf , I will merge this as is, can you prepare a PR with the discussed modification and additional tests to cover that additional scenario please?

Thanks

Kevin

Reply:
Hey Kevin, thanks for merging this.
I can circle back to this in the future, @auduchinok any specific examples you can think of to cover this?

Reply:
@nojaf Nope, I couldn't easily find an example. I'm going to look at members parser recovery later, and maybe I'll find an example. What happened in #12224 was it started returning intersecting ranges only after a better recovery was added to match clauses and patterns parsing, so it could probably happen here too.